packages/actions/src/internal/runCallWithPrestateTrace.js

```diff
@@ -0,0 +1,203 @@
+		const storage = {}
+		for (const slotHex of slots) {
+			try {
+				// Convert hex slot to Uint8Array for getContractStorage
+				const slotKey = Buffer.from(slotHex.slice(2), 'hex')
+				const value = await vm.stateManager.getContractStorage(address, slotKey)
+				storage[slotHex] = bytesToHex(value)
+			} catch (err) {
+				logger.error(err, `Error getting storage at slot ${slotHex} for account ${address.toString()}`)
+			}
+		}
+
+		return {
+			balance: toHex(account.balance),
+			nonce: account.nonce.toString(),
+			code: code && code.length > 0 ? bytesToHex(code) : '0x',
+			storage,
+		}
+	} catch (err) {
+		logger.error(err, `Error capturing state for account ${address.toString()}`)
+		return null
+	}
+}
+
+/**
+ * @internal
+ * Executes a call with prestate tracer, which captures account state before and after execution
+ * @template {boolean} TDiffMode
+ * @param {import('@tevm/vm').Vm} vm
+ * @param {import('@tevm/node').TevmNode['logger']} logger
+ * @param {import('@tevm/evm').EvmRunCallOpts} params
+ * @param {TDiffMode} diffMode If true, only returns state that changed between pre and post execution
+ * @returns {Promise<import('@tevm/evm').EvmResult & {trace: TDiffMode extends true ? import('../debug/DebugResult.js').PrestateTracerDiffResult : import('../debug/DebugResult.js').PrestateTracerResult}>}
+ * @throws {never}
+ */
+export const runCallWithPrestateTrace = async (vm, logger, params, diffMode = /** @type {TDiffMode} */ (false)) => {
+	// Prepare VM for tracing
+	await vm.evm.journal.cleanup()
+	// Start tracking which accounts and storage slots are accessed during execution
+	vm.evm.journal.startReportingAccessList()
+
+	/** @type {any} */
+	let trace
+
+	// Clone the state manager to capture state before execution
+	const vmClone = await vm.deepCopy()
+
+	// Run the call
+	logger.debug(params, 'runCallWithPrestateTrace: executing call')
+	const runCallResult = await vm.evm.runCall(params)
+
+	// Get the access list to determine which accounts and storage slots were accessed
+	if (!vm.evm.journal.accessList) {
+		throw new DefensiveNullCheckError('Expected journal access list to be defined')
+	}
+
+	// Transform access list to usable format
+	/** @type {Map<import('@tevm/utils').Hex, Set<import('@tevm/utils').Hex>>} */
+	const accessListMap = new Map()
+	for (const [addressString, slots] of vm.evm.journal.accessList) {
+		const slotSet = new Set()
+		for (const slotString of slots) slotSet.add(`0x${slotString}`)
+		accessListMap.set(`0x${addressString}`, slotSet)
+	}
+
+	// Capture pre-state (from cloned VM before execution)
+	/** @type {import('../debug/DebugResult.js').PrestateTracerResult} */
+	const preState = {}
+	for (const [address] of accessListMap) {
+		try {
+			const ethAddress = createAddress(address)
+			const slots = accessListMap.get(address) || new Set()
+			const accountState = await captureAccountState(vmClone, logger, ethAddress, slots)
```

Comment:
```suggestion
			const accountState = await captureAccountState(vmClone, logger, ethAddress, slots)
      for slot in theaccesslist {
        const preimage = vm.evm.journal.preimages.get(keccak256(preimage))
      }
```
This is an easier way to get the diff using built in statemanager stuff